### ISSUE TITLE
Deep network gradient compression.

### ISSUE THREAD

**Issue by gongweibao** [PaddlePaddle/Paddle#15021]
- [ ] Mixed precision.
   - [ ] On Bert.
   - [ ] On Resnet 50 on imagenet dataset.
- [x] DGC
   - [x] Accumulate local gradients < threshold and send them when > threshold.
      - [x] nccl
           - [x] AllReduce
           - ~~[ ] Reduce, broadcast~~
      - ~~[ ] pserver~~
   - [x] Warm up
   - [x] momentum correction.
   - [x] local gradient clipping.
   - [x] momentum factor masking and warmup training to overcome the staleness problem caused by reduced communication.
